core/src/main/java/org/mule/runtime/core/routing/Splitter.java

```diff
@@ -9,26 +9,43 @@
 import org.mule.runtime.api.lifecycle.Initialisable;
 import org.mule.runtime.api.lifecycle.InitialisationException;
 import org.mule.runtime.core.api.Event;
+import org.mule.runtime.core.expression.ExpressionConfig;
 import org.mule.runtime.core.routing.outbound.AbstractMessageSequenceSplitter;
 import org.mule.runtime.core.util.collection.EventToMessageSequenceSplittingStrategy;
 import org.mule.runtime.core.util.collection.SplittingStrategy;
 
 /**
- * Splits a message that has a Collection, Iterable, MessageSequence or Iterator payload invoking the next message processor one
- * for each item in it.
+ * Splits a message that has a Collection, Iterable, MessageSequence or Iterator payload or an expression that resolves to some of
+ * those types or data that is a collection of values in a non-java format. Then invokes the next message processor one for each
+ * item in it.
  * <p>
  * <b>EIP Reference:</b> <a href="http://www.eaipatterns.com/Sequencer.html">http ://www.eaipatterns.com/Sequencer.html</a>
  */
-public class CollectionSplitter extends AbstractMessageSequenceSplitter implements Initialisable {
+public class Splitter extends AbstractMessageSequenceSplitter implements Initialisable {
```

Comment:
There's no AbstractMessageSequenceSplitter now. Yes, I have consider doing splitter a scope but it was not in the scope of this task to do it. We can create a issue for that as a NTH.